src/mesh/logical_location.hpp

```diff
@@ -244,6 +264,64 @@ DetermineOwnership(const LogicalLocation &main_block,
   return main_owns;
 }
 
+// Given a topological element, ownership array of the sending block, and offset indices
+// defining the location of an index region within the block (i.e. the ghost zones passed
+// across the x-face or the ghost zones passed across the z-edge), return the index range
+// masking array required for masking out unowned regions of the index space. ox? defines
+// buffer location on the owner block
+inline auto GetIndexRangeMaskFromOwnership(TopologicalElement el,
+                                           const block_ownership_t &sender_ownership,
+                                           int ox1, int ox2, int ox3) {
```

Comment:
This seems like it was a bit tricky to work out but the resultant code is pretty clean. I like it.

Reply:
Getting to this point definitely required a bit of thought.